### PR TITLE
Add an extra test case for mapped type relationships

### PR DESCRIPTION
**Pull request by Andarist**
closes https://github.com/microsoft/TypeScript/issues/32365
it was fixed by https://github.com/microsoft/TypeScript/pull/32071

### CODE REVIEW

File: tests/cases/conformance/types/mapped/mappedTypeRelationships.ts [hunk 1]
```diff
@@ -192,3 +192,21 @@ function f90<T extends { x: number }>() {
...
     return undefined! as T;
 }
+
+// #32365
+interface SettingsTypes {
+  audio: {
+    volume: string;
+  };
+  video: {
+    resulution: string;
```
Comment (DanielRosenwasser) [inline tests/cases/conformance/types/mapped/mappedTypeRelationships.ts:202]: I know it was part of the original issue - but can you fix this to `resolution`?

File: tests/cases/conformance/types/mapped/mappedTypeRelationships.ts [hunk 2]
```diff
@@ -192,3 +192,21 @@ function f90<T extends { x: number }>() {
...
+  };
+}
+interface Settings<Params extends { [K in keyof Params]?: string }> {
+  config: Params;
+}
+type ConcreteSettingsResult1 = Settings<SettingsTypes["audio"]>;
+type ConcreteSettingsResult2 = Settings<SettingsTypes["audio" | 'video']>;
```
Comment (DanielRosenwasser) [inline tests/cases/conformance/types/mapped/mappedTypeRelationships.ts:209]: Let's use a single quote style here.

Comment (fatcerberus) [reply tests/cases/conformance/types/mapped/mappedTypeRelationships.ts:209]: Should one read this as "(single quote) style" or "single (quote style)" 🚎

Comment (jakebailey) [reply tests/cases/conformance/types/mapped/mappedTypeRelationships.ts:209]: The latter, e.g. just use double quotes.

**Comment by Andarist** [comment]
@DanielRosenwasser @jakebailey addressed both nits

**Review by jakebailey — APPROVED** [review]

**Pull request merged**